docs/src/manual/variables.md

```diff
@@ -589,178 +908,82 @@ model[:x]
-```jldoctest variables_binary
-julia> is_binary(x)
-true
-
-julia> unset_binary(x)
-
-julia> is_binary(x)
-false
-```
-
-Binary optimization variables can also be created by setting the `binary`
-keyword to `true`.
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x, binary=true)
-x
-```
-### Integer constraints
-
-Integer optimization variables are constrained to the set ``x \in \mathbb{Z}``.
-(The `MOI.Integer` set in MathOptInterface.) Integer optimization variables can
-be created in JuMP by passing `Int` as an optional positional argument:
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x, Int)
-x
-```
-Integer optimization variables can also be created by setting the `integer`
-keyword to `true`.
-```jldoctest variables_integer; setup=:(model=Model())
-julia> @variable(model, x, integer=true)
-x
-```
-We can check if an optimization variable is integer by calling
-[`is_integer`](@ref) on the JuMP variable, and integer constraints can be
-removed with [`unset_integer`](@ref).
-```jldoctest variables_integer
-julia> is_integer(x)
-true
-
-julia> unset_integer(x)
-
-julia> is_integer(x)
-false
-```
-
-!!! tip
-    The [`relax_integrality`](@ref) function relaxes all integrality constraints
-    in the model, returning a function that can be called to undo the operation
-    later on.
-
 ## Semidefinite variables
 
-JuMP also supports modeling with semidefinite variables. A square symmetric
-matrix ``X`` is positive semidefinite if all eigenvalues are nonnegative. We can
-declare a matrix of JuMP variables to be positive semidefinite as follows:
+A square symmetric matrix ``X`` is positive semidefinite if all eigenvalues are
+nonnegative.
+
+Declare a matrix of JuMP variables to be positive semidefinite by passing `PSD`
+as an optional positional argument:
 ```jldoctest; setup=:(model=Model())
 julia> @variable(model, x[1:2, 1:2], PSD)
 2×2 LinearAlgebra.Symmetric{VariableRef, Matrix{VariableRef}}:
  x[1,1]  x[1,2]
  x[1,2]  x[2,2]
 ```
-or using the syntax for [Variables constrained on creation](@ref jump_variables_on_creation):
-```jldoctest; setup=:(model=Model())
-julia> @variable(model, x[1:2, 1:2] in PSDCone())
-2×2 LinearAlgebra.Symmetric{VariableRef, Matrix{VariableRef}}:
- x[1,1]  x[1,2]
- x[1,2]  x[2,2]
-```
 
-Note that `x` must be a square 2-dimensional `Array` of JuMP variables; it
-cannot be a DenseAxisArray or a SparseAxisArray.
-(See [Variable containers](@ref), above, for more on this.)
+!!! note
+    `x` must be a square 2-dimensional `Array` of JuMP variables; it cannot be a
+    DenseAxisArray or a SparseAxisArray.
+
+## Symmetric variables
 
-You can also impose a weaker constraint that the square matrix is only symmetric
-(instead of positive semidefinite) as follows:
+Declare a square matrix of JuMP variables to be symmetric by passing
+`Symmetric`  as an optional positional argument:
 ```jldoctest; setup=:(model=Model())
 julia> @variable(model, x[1:2, 1:2], Symmetric)
```

Comment:
This can also be done with `@constraint(model, x[1:2, 1:2] in SymMatrixSpace())`

Reply:
Added an example.